### PR TITLE
Add observers to user defaults

### PR DESCRIPTION
**Pull request by ibeckermayer**
Added observers to make `Home` respond to changes in `UserDefaults.shared.didUserMakeContactWithSickUser` and `UserDefaults.shared.lastTestedDate`. Additional changes to the `drawScreen()` were necessary to prevent redrawing / other funkiness on state change.

Linter checks are going to be complaining over the TODO's I left in there, which are a result of my incredibly jank method for testing this: I've set it up so that when you click the `spreadButton` it logically not's the `UserDefaults.shared.didUserMakeContactWithSickUser` value and makes the user have been tested 15 days ago, and when you click the `testedButton` it makes the user have been tested 13 days ago. This way you can manually check all the different transitions work properly (just keep clicking the buttons).

You can test that manually yourself, once this PR is approved I'll delete that logic and uncomment the segue logic.

### CODE REVIEW

File: COVIDWatch iOS/ViewControllers/Home.swift [hunk 1]
```diff
@@ -17,12 +17,28 @@ class Home: BaseViewController {
...
+            changeHandler: { (_, _) in
+            self.drawScreen()
+        })
+
+        didUserMakeContactWithSickUserObserver = UserDefaults.shared.observe(
+            \.didUserMakeContactWithSickUser,
+            options: [],
+            changeHandler: { (_, _) in
+            self.drawScreen()
+        })
```
Comment (jeffctown) [inline COVIDWatch iOS/ViewControllers/Home.swift:39]: Does observation registration need to happen when the views are laid out (which can happen a LOT)? Or can this be moved to viewDidLoad (which only happens once)?

**Review by jeffctown — COMMENTED** [review]
Looks fine overall. Just one question.

File: COVIDWatch iOS/ViewControllers/Home.swift (hunk 1 above)
Comment (ibeckermayer) [reply COVIDWatch iOS/ViewControllers/Home.swift:39]: Good point, it only needs to happen once.

**Comment by therealnikhil** [comment]
"I've set it up so that when you click the spreadButton it logically not's the UserDefaults.shared.didUserMakeContactWithSickUser value and makes the user have been tested 15 days ago, and when you click the testedButton it makes the user have been tested 13 days ago." Just to be sure, this logic is just for testing, yes?

**Comment by ibeckermayer** [comment]
> "I've set it up so that when you click the spreadButton it logically not's the UserDefaults.shared.didUserMakeContactWithSickUser value and makes the user have been tested 15 days ago, and when you click the testedButton it makes the user have been tested 13 days ago." Just to be sure, this logic is just for testing, yes?

Yes that was just for testing and now its gone. For now you'll have to take my word that I tested it; it maybe makes sense to create a ticket that adds some options in the menu to toggle these states manually based on a `DEV` flag.

**Comment by therealnikhil** [comment]
Sounds good, we don't want that code anyway to be on dev.

**Review by therealnikhil — APPROVED** [review]
Looks good!

**Pull request merged**